### PR TITLE
Fix intermediate commit for arangosearch links and inverted index

### PR DESCRIPTION
**Pull request by MBkkt**
### Scope & Purpose

How it done:
1. We shouldn't call any callbacks for abort/commit fail/etc
2. Store callbacks in TransactionState
3. Move it call to internal part of State, because of function `performIntermediateCommitIfRequired`

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.10: *(Please link PR)*
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: tests/js/client/chaos/test-chaos-load-common.inc [hunk 1]
```diff
@@ -404,17 +407,17 @@ function BaseChaosSuite(testOpts) {
...
               c.truncate();
             } else if (d >= 0.9) {
-              let o = opts();
               let d = docs();
+              let o = opts(d.length);
               log(`RUNNING AQL INSERT WITH ${d.length} DOCS. OPTIONS: ${JSON.stringify(o)}`);
               query("FOR doc IN @docs INSERT doc INTO " + c.name(), {docs: d}, o);
             } else if (d >= 0.8) {
-              let o = opts();
+              let o = opts(1);
```
Comment (jsteemann) [inline tests/js/client/chaos/test-chaos-load-common.inc:415]: Should this be `opts(limit)`?

File: tests/js/client/chaos/test-chaos-load-common.inc [hunk 2]
```diff
@@ -404,17 +407,17 @@ function BaseChaosSuite(testOpts) {
...
               const limit = Math.floor(Math.random() * 200);
               log(`RUNNING AQL REMOVE WITH LIMIT=${limit}. OPTIONS: ${JSON.stringify(o)}`);
               query("FOR doc IN " + c.name() + " LIMIT @limit REMOVE doc IN " + c.name(), {limit}, o);
             } else if (d >= 0.7) {
-              let o = opts();
+              let o = opts(1);
```
Comment (jsteemann) [inline tests/js/client/chaos/test-chaos-load-common.inc:420]: Should this be `opts(limit)`?

File: tests/js/client/chaos/test-chaos-load-common.inc [hunk 3]
```diff
@@ -446,13 +449,13 @@ function BaseChaosSuite(testOpts) {
                 }
               }
             } else if (d >= 0.65) {
-              let o = opts();
+              let o = opts(1);
```
Comment (jsteemann) [inline tests/js/client/chaos/test-chaos-load-common.inc:452]: Should this be `opts(limit)`?

File: arangod/IResearch/IResearchDataStore.h [hunk 4]
```diff
@@ -65,38 +65,24 @@ struct IResearchTrxState final : public TransactionState::Cookie {
...
-          << e.what();
-    } catch (...) {
-      LOG_TOPIC("14917", ERR, arangodb::iresearch::TOPIC)
-          << "caught exception while applying accumulated removals";
+    if (!_wasCommit) {
+      _removals.clear();
+      _ctx.reset();
+    } else {
+      TRI_ASSERT(_removals.empty());
     }
```
Comment (jsteemann) [inline arangod/IResearch/IResearchDataStore.h:79]:
```suggestion
    }
    TRI_ASSERT(_removals.empty());
```

File: tests/js/client/chaos/test-chaos-load-common.inc (hunk 3 above)
Comment (MBkkt) [reply tests/js/client/chaos/test-chaos-load-common.inc:452]: True, thanks

File: arangod/IResearch/IResearchDataStore.h (hunk 4 above)
Comment (MBkkt) [reply arangod/IResearch/IResearchDataStore.h]: Done

**Review by Dronplane — APPROVED** [review]
LGTM

**Review by jsteemann — APPROVED** [review]
LGTM!

**Review by gnusi — APPROVED** [review]

**Pull request merged**